src/main/java/seedu/address/model/ReadOnlyFocusMode.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.model;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Unmodifiable view of the state of the focus mode.
+ */
+public interface ReadOnlyFocusMode {
+    /**
+     * TODO: replace with class
+     * Enters into focus mode for the given class.
+     *
+     * @param classToFocus Class corresponding to the focus mode.
+     */
+    void enter(Tag classToFocus);
+
+    /**
+     * Exits focus mode.
+     */
+    void exit();
+
+    /**
+     * TODO: replace with class
+     * Sets a new class to focus on.
+     *
+     * @param classToFocus Class corresponding to the focus mode.
+     */
+    void setFocusedClass(Tag classToFocus);
```

Comment:
Since the interface's intention is such that the focus mode is read only, should we still allow it to set the focused class?